tests/cpp/nested_msg_test.cc

```diff
@@ -186,3 +204,34 @@ TEST(NestedMsg, CombineNestedMsg) {
   EXPECT_TRUE(Equal(output, expected,
                     [](Integer lhs, Integer rhs) -> bool { return lhs->value == rhs->value; }));
 }
+
+TEST(NestedMsg, TransformTupleLeaf) {
+  auto c0 = Integer(0);
```

Comment:
Add a case to check that you are indeed returning the orginal tuple when part of the values does not change.
Please try to run this test locally